docscan/session/create/check/liveness.go

```diff
@@ -36,34 +51,34 @@ func NewRequestedLivenessCheckBuilder() *RequestedLivenessCheckBuilder {
 type RequestedLivenessCheckBuilder struct {
```

Comment:
I am not a huge fan of this builder structure. I'm not sure why purpose it serves.

Why not simply do:
```
func NewRequestedLivenessCheck(c *RequestedLivenessConfig) *RequestedLivenessCheck {
   return &RequestedLivenessCheck{
		checkType: constants.Liveness,
		config:    config,
	}
}
```

Reply:
We've gone with builders for all of the Doc Scan payloads so that we could add new items to the payload, e.g. we would have a second argument in the function you specify:
```go
func NewRequestedLivenessCheck(c *RequestedLivenessConfig, priority int) *RequestedLivenessCheck {
   return &RequestedLivenessCheck{
		checkType: constants.Liveness,
		config:    config,
                priority:   priority
	}
}
```

Not a great example, as it could go in config, but the principle of adding/removing items from the constructor is hidden from the user (if they use it as intended), if they use the builder. And so we can make & break, without worrying about backwards compatibility

Reply:
yeah, I would think it would be preferable to hold those things in the Config. 

as for default values, you could still have that:
```
func NewRequestedLivenessCheck(c *RequestedLivenessConfig) *RequestedLivenessCheck {
   priority := c.Priority
   if priority == PriorityUndefined {
      priority = PriorityDefault
   }
   return &RequestedLivenessCheck{
		checkType: constants.Liveness,
		config:    config,
                priority: priority,
	}
}
```

Reply:
So looking a bit closer, I think maybe the issue here is that `RequestedLivenessConfig` is supposed to marshal directly into the config in the structure send over the wire. Essentially your builder struct is the actual config from the integraters POV.

This is probably OK actually, but I would drop the setters, and just keep the values exported so the integrator can do:
```
   builder := &RequestedLivenessCheckBuilder{
     LivenessType: ZoomLivenessType,
     MaxRetries: 5,
  }
```

Reply:
I think I'm withdrawing my objections here. I have some minor concerns around int as a pointer and a suggestion to type alias the const values, but we're addressing that elsewhere.